compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java

```diff
@@ -1532,7 +1532,7 @@ public void visit(BLangRecordLiteral.BLangRecordKeyValueField recordKeyValue) {
 
     @Override
     public void visit(BLangRecordLiteral.BLangRecordKey recordKey) {
-        if (setEnclosingNode(recordKey.fieldSymbol, recordKey.pos)) {
+        if (recordKey.fieldSymbol != null && setEnclosingNode(recordKey.fieldSymbol, recordKey.pos)) {
```

Comment:
Why is this null check needed?

Reply:
Yes, I also wanted to check from Anjana. Do we need this change with other fix you have done here. @sanjana

Reply:
@pubudu91 @mohanvive 

The null check here is essential. 

If the field has a computed key, the `fieldSymbol` is set to `null`.

https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L6556-L6570

Therefore, this null check is required to bypass this `if` condition.

Reply:
Another possible approach would be as following: 

`if (!recordKey.computedKey && setEnclosingNode(recordKey.fieldSymbol, recordKey.pos))`

Reply:
Ah I see. I think the second approach you suggested would be better since it's more clear as to what we are checking there.

Reply:
Updated.